crates/context_aware_config/src/api/config/handlers.rs

```diff
@@ -297,23 +298,27 @@ fn get_contextids_from_overrideid(
     Ok(res)
 }
 
-fn construct_new_payload(req_payload: &Map<String, Value>) -> web::Json<PutReq> {
+fn construct_new_payload(
+    req_payload: &Map<String, Value>,
+) -> superposition::Result<web::Json<PutReq>> {
     let mut res = req_payload.clone();
     res.remove("to_be_deleted");
     res.remove("override_id");
     res.remove("id");
     if let Some(Value::Object(res_context)) = res.get("context") {
```

Comment:
Can we do negative checks first to avoid this nesting?
```suggestion
    if res.get("context").is_none() {
```
Throw error if true, process if false